package fantasybit;

message SettlePos {
    optional string pk = 10;
    optional int32 qty = 20;
    optional int32 price = 25;
}

message BookPos {
    optional string playerid = 30;
    repeated SettlePos positions = 40;
}

message GameSettlePos {
    optional string gameid = 10;
    repeated BookPos home = 20;
    repeated BookPos away = 30;
}

message OrderCore {
    optional bool buyside = 10;
    optional int32 size = 20;
    optional int32 price = 30;
}

message Order {
    optional int32 refnum = 10;
    optional OrderCore core = 20;
}

message MarketTicker {
    enum Type { BID=1; ASK=2; LAST=3; }

    optional Type type = 1;
    optional int32 size = 20;
    optional int32 price = 30;
}

message DepthItem {
    optional int32 level = 10;
    optional int32 b = 30;
    optional int32 a = 40;
    optional int32 bs = 50;
    optional int32 as = 60;

}

message BookDelta {
    optional string public_key = 10;
    optional int32 seqnum = 20;
    optional string playerid = 30;
    optional OrderCore newnew = 40;
    repeated Order removes = 50;
    repeated MarketTicker level2tic = 60;
    repeated MarketTicker level1tic = 70;
}

message MarketSnapshot {
    optional string symbol = 10;
    repeated DepthItem depth = 70;
}
